common/scala/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerFactory.scala

```diff
@@ -52,7 +52,23 @@ case class ContainerPoolConfig(userMemory: ByteSize, concurrentPeekFactor: Doubl
  */
 trait ContainerFactory {
 
-  /** create a new Container */
+  /**
+   * Create a new Container
+   *
+   * The created container has to satisfy following requirements:
+   * - The container's file system is based on the provided action image and has a read/write layer on top.
+   * - If the specified image is not available on the system, it is pulled from an image
+   *   repository - for example, Docker Hub.
+   * - The container has a network interface which is able to connect to the internet
```

Comment:
Do we require containers to be able to connect to the internet?

We could instead state that we don't do any work to connect a container to the internet. It either is connected already or it isn't. DNS servers seem not to be known to this interface.

Reply:
DNS server settings are part of the the `ContainerArgsConfig`. At least, the container must be able to connect to the DNS servers defined there.

What about this proposal?

_The container needs a network setup - usually, a network interface - such that the invoker is able to connect the action container. The container must be able to perform DNS resolution based on the settings provided via `ContainerArgsConfig`._ If needed by action authors, the container should be able to connect to other systems or even the internet to consume services._

Reply:
Add an "if any" after `ContainerArgsConfig`, then lgtm :)

Reply:
Updated as discussed.